libnmstate/nm/applier.py

```diff
@@ -413,14 +413,12 @@ def _build_connection_profile(iface_desired_state, base_con_profile=None):
     if bond_opts:
         settings.append(bond.create_setting(bond_opts, wired_setting))
     elif iface_type == bridge.BRIDGE_TYPE:
-        bridge_options = iface_desired_state.get(bridge.BRIDGE_TYPE, {}).get(
-            LB.OPTIONS_SUBTREE
+        bridge_state = iface_desired_state.get(bridge.BRIDGE_TYPE, {})
+        bridge_options = bridge_state.get(LB.OPTIONS_SUBTREE)
+        linux_bridge_setting = bridge.create_setting(
+            bridge_options, base_profile, iface_desired_state,
```

Comment:
The `iface_desired_state` argument already contains the information of `bridge_options`.

Passing `iface_desired_state` and `base_profile` to `bridge.create_setting()` is enough.

Reply:
While you're 100% right, this will lead to having to refactor the `linux_bridge_test.py` module.

I'm OK with doing it - just want everyone to be aware this PR will keep on growing,